src/common/hashes/index.ts

```diff
@@ -59,13 +61,31 @@ const addLengthPrefix = (hex: string): string => {
   throw new Error('Variable integer overflow.')
 }
 
+/**
+ * Hashes the Transaction blob as the ledger does. Only valid for signed Transaction blobs.
+ * 
+ * @param {Transaction} tx A blob of a signed Transaction to hash.
+ * @returns {string} The hash of tx
+ */
+
 export const computeBinaryTransactionHash = (txBlobHex: string): string => {
   const prefix = HashPrefix.TRANSACTION_ID.toString(16).toUpperCase()
-  return sha512Half(prefix + txBlobHex)
+  return sha512Half(prefix.concat(txBlobHex))
 }
 
-export const computeTransactionHash = (txJSON: any): string => {
-  return computeBinaryTransactionHash(encode(txJSON))
+/**
+ * Hashes the Transaction object as the ledger does. Throws if the transaction is unsigned.
+ * 
+ * @param {Transaction} tx A transaction to hash. Tx must be signed.
+ * @returns {string} A hash of tx
+ * @throws {ValidationError} if the Transaction is unsigned
+ */
+
+export const computeSignedTransactionHash = (tx: Transaction): string => {
```

Comment:
I would combine `computeSignedTransactionHash` and `computeBinaryTransactionHash`, and have `tx` be either a string or a `Transaction`. Then you can just check the type and encode if needed.